### ISSUE TITLE
Reconcile envoyproxy/envoy and envoyproxy/data-plane-api

### ISSUE THREAD

**Issue by htuch** [envoyproxy/envoy#2934]
As discussed at this week's community call, we would like to find a way to improve velocity and reduce futz when making changes which involve release notes, docs and API, while still preserving the standalone capability of `data-plane-api` as the canonical API reference.

The proposal is to do something similar to what Istio have recently proposed (https://groups.google.com/d/msg/istio-dev/znLqMD-j-4o/AGvNAN2cAQAJ).  We will have APIs and docs in  `envoyproxy/envoy`, and then run a nightly job or a webhook bot to copy the API protos to `envoyproxy/data-plane-api`. 

This will allow all the docs, release notes and API changes to be made alongside the implementation work in the main repo. This should not impact any existing users of the API protos.

We should ensure that the repo sync includes some SHA to allow consumers to be aware of the relationship between `envoyproxy/data-plane-api` commits and `envoyproxy/envoy`

**Comment by ggreenway** [comment]
Instead of a nightly or time-based job, can we do it as a build step on each commit? Also, if we're doing it for each commit, we can more easily take the commit message and replicate it in the data-plane-api generated commit, so at least there's some meaningful history to read in that repo.

**Comment by mattklein123** [comment]
Yes, per @ggreenway we can just do this is a master CI job, much like we do docs, etc. It's pretty easy to push commits via Circle. I think that's a great idea.

**Comment by htuch** [comment]
https://github.com/envoyproxy/envoy/pull/3036 is ready to go. As per https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/envoy-dev/KcVHFH-zQwQ, we intend to land this today.
